bot/worker.py

```diff
@@ -0,0 +1,184 @@
+    if ' ' not in processed_body:
+        return
+
+    if processed_body in config.EXCLUDED_RESPONSES:
+        return
+
+    if processed_body in config.CUSTOM_RESPONSES:
+        do_custom_reply(thing=thing, custom_response=config.CUSTOM_RESPONSES[processed_body])
+
+    if do_flair_specific_reply(thing, processed_body):
+        return
+
+    do_regular_reply(thing, processed_body)
+
+
+def process_body(body_text):
+    """Method used to clean the thing text. Logic is similar to clean_response_text on wiki parsers.
+    * If thing contains a quote, the first quote is considered as the response_text.
+    * Punctuation marks are replaced with space.
+    * The response_text is turned to lowercase.
+    * Converts multiple spaces into single space.
+
+    Removed code to remove repeating letters in a thing because it does more harm than good - words like 'all',
+    'tree' are stripped to 'al' and 'tre' which dont match with any responses.
+
+    :param body_text: The thing body
+    :return: Processed thing body
+    """
+
+    if '>' in body_text:
+        lines = body_text.split('\n\n')
+        for line in lines:
+            if line.startswith('>'):
+                body_text = line
+                break
+
+    body_text = body_text.translate(PUNCTUATION_TRANS)
+    body_text = body_text.translate(WHITESPACE_TRANS)
+
+    while '  ' in body_text:
+        body_text = body_text.replace('  ', ' ')
+
+    body_text = body_text.strip().lower()
+
+    return body_text
+
+
+def do_flair_specific_reply(thing, response):
+    hero_id = db_api.get_hero_id_by_flair_css(flair_css=thing.author_flair_css_class)
+    if hero_id:
+        link, hero_id = db_api.get_link_for_response(
+            response=response, hero_id=hero_id)
+        if link:
+            reply = create_reply(thing=thing, response_url=link, hero_id=hero_id)
+            thing.reply(reply)
+            logger.info("Added: " + thing.fullname)
+            return True
+
+
+def do_regular_reply(thing, response):
+    """Method to create response for given thing.
+    In case of multiple matches, it used to sort responses in descending order of heroes, but now it's random.
+    add_shitty_wizard_response was very similar to this, and hence has been merged
+
+    :param thing: The thing/submission on reddit
+    :param response: The plaintext processed thing body
+    :return: None
+    """
+
+    link, hero_id = db_api.get_link_for_response(processed_text=response)
+
+    if link and hero_id:
+        img_dir = db_api.get_img_dir_by_id(hero_id=hero_id)
+
+        thing.reply(create_reply(thing=thing, response_url=link, hero_id=hero_id, img=img_dir))
+
+        logger.info("Replied to: " + thing.id)
+
+
+def create_reply(thing, response_url, hero_id, img=None):
+    """Method that creates a reply in reddit format.
+
+    The message consists of a link the the response, the response itself, a warning about the sound
```

Comment:
"a link the the response" should probably be "a link to the response".

Reply:
Fixed in 41e1416fd2c7051547d1ef4807ac17741893a239